### PR TITLE
BC-7101 - collaboration service deployment

### PR DESCRIPTION
**Pull request by hoeppner-dataport**
# Description
Implement a deployment for the BoardCollaboration-WebSocket-Server.

## Links to Tickets or other pull requests
https://ticketsystem.dbildungscloud.de/browse/BC-7101
server: https://github.com/hpi-schul-cloud/schulcloud-server/pull/4982
vue: https://github.com/hpi-schul-cloud/nuxt-client/pull/3234

## Approval for review

- [ ] All points were discussed with the ticket creator, support-team or product owner. The code upholds all quality guidelines from the PR-template.

> Notice: Please remove the WIP label if the PR is ready to review, otherwise nobody will review it.

### CODE REVIEW

File: ansible/group_vars/develop/cfg.yml [hunk 1]
```diff
@@ -34,3 +34,6 @@ TLDRAW_SERVER_REPLICAS: 3
 ADMIN_API_SERVER_REPLICAS: 1
 
 SC_DEFAULT_STORAGE_CLASS_NAME: "nfs-client"
+
+FEATURE_COLUMN_BOARD_SOCKET_ENABLED: "true"
+BOARD_COLLABORATION_SERVER_REPLICAS: 3
```
Comment (Metauriel) [inline ansible/group_vars/develop/cfg.yml:39]: that wont work until we have done the ticket for scaling...

Comment (hoeppner-dataport) [reply ansible/group_vars/develop/cfg.yml]: Changed to one replica

**Review by Metauriel — APPROVED** [review]

File: ansible/group_vars/develop/cfg.yml [hunk 2]
```diff
@@ -35,4 +35,10 @@ ADMIN_API_SERVER_REPLICAS: 1
 
 SC_DEFAULT_STORAGE_CLASS_NAME: "nfs-client"
 
+FEATURE_COLUMN_BOARD_SOCKET_ENABLED: "true"
+
+BOARD_COLLABORATION_SERVER_REPLICAS: 1
+
+BOARD_COLLABORATION_URI: "ws://localhost:4450"
```
Comment (mamutmk5) [inline ansible/group_vars/develop/cfg.yml:42]: If this works, then the websocet will be bound on the pod to the pod and not for any remote system or via the deployment, please chack this and myebe use {{ DOMAIN }} instant as host and no port.

File: ansible/group_vars/all/config.yml [hunk 3]
```diff
@@ -1049,6 +1049,11 @@ configuration_all:
     server: true
     client: false
     nuxtclient: false
+  BOARD_COLLABORATION_URI:
+    value: ""
```
Comment (mamutmk5) [inline ansible/group_vars/all/config.yml:1053]: Only add an default value if you want to set them to an default value who is not empty and "" is to see as empty
mabey set the websocet to the domain is an good idee
```suggestion
    value: "ws://{{ DOMAIN }}"
```

**Review by mamutmk5 — CHANGES_REQUESTED** [review]

**Review by mamutmk5 — APPROVED** [review]

File: ansible/group_vars/develop/cfg.yml (hunk 2 above)
Comment (hoeppner-dataport) [reply ansible/group_vars/develop/cfg.yml]: removed the whole setting

File: ansible/group_vars/all/config.yml (hunk 3 above)
Comment (hoeppner-dataport) [reply ansible/group_vars/all/config.yml]: fixed as suggested

**Pull request merged**